### DISCUSSION TITLE
What is causing this rerender?

### DISCUSSION THREAD

**Discussion by Undistraction** [gatsbyjs/gatsby#29733] [Help]
I'm trying to understand why opening a modal causes a full page rerender. Using the profiler I'm seeing the following: 

<img width="595" alt="Screenshot 2021-02-24 at 11 25 20" src="https://user-images.githubusercontent.com/161489/108994042-38ca8a80-7693-11eb-9c9b-15a3a8985a5a.png">

I know `AppContainer` is part of Gatsby's internals, and I'm assuming `_default` is too, but why is `_default` rerendering? The reason given is that its parent props have changed, however, `AppContainer` isn't rerendering. What props could have changed in `AppContainer` that it would pass into `_default`?

**Comment by herecydev** [comment]
PageRenderer is really the last part of "Gatsby internals", are you sure `AppContainer` isn't something you're bringing in? Perhaps as part of `wrapPageElement`?
